azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/Account.java

```diff
@@ -150,54 +164,115 @@ public String toString() {
+                credential.getToken(tokenRequestContext);
+
+        return mono.flatMap(accessToken ->
+                Mono.zip(AzureResourceManager.authenticate(request -> Mono.just(accessToken), new AzureProfile(env))
+                        .tenants().listAsync().map(Tenant::tenantId).collectList(), Mono.just(accessToken)));
     }
 
-    protected void initializeSubscriptions() {
-        List<Subscription> subscriptions = listSubscriptions(entity.getTenantIds(), entity.getTenantCredential());
-        entity.setTenantIds(subscriptions.stream().map(Subscription::getTenantId).distinct().collect(Collectors.toList()));
-        entity.setSubscriptions(subscriptions);
-        selectSubscriptionInner(subscriptions, this.entity.getSelectedSubscriptionIds());
+    protected Mono<Void> initializeSubscriptions() {
+        return listSubscriptions(entity.getTenantIds(), entity.getTenantCredential()).doOnSuccess(subscriptions -> {
+            entity.setTenantIds(subscriptions.stream().map(Subscription::getTenantId).distinct().collect(Collectors.toList()));
+            entity.setSubscriptions(subscriptions);
+            selectSubscriptionInner(subscriptions, this.entity.getSelectedSubscriptionIds());
 
-        // no subscriptions are selected, selected all
-        if (!subscriptions.isEmpty() && subscriptions.stream().noneMatch(Subscription::isSelected)) {
-            subscriptions.forEach(s -> s.setSelected(true));
-        }
+            // no subscriptions are selected, selected all
+            if (!subscriptions.isEmpty() && subscriptions.stream().noneMatch(Subscription::isSelected)) {
+                subscriptions.forEach(s -> s.setSelected(true));
+            }
+        }).then();
+    }
+
+    private Mono<List<Subscription>> listSubscriptions(List<String> tenantIds, TenantCredential credential) {
+        AzureProfile azureProfile = new AzureProfile(ObjectUtils.firstNonNull(this.entity.getEnvironment(), Azure.az(AzureCloud.class).getOrDefault()));
+        // use map to re-dup subs
+        final Map<String, Subscription> subscriptionMap = new ConcurrentHashMap<>();
+        Map<String, Throwable> throwableList = new ConcurrentHashMap<>();
+        return Flux.fromIterable(tenantIds).parallel().runOn(Schedulers.boundedElastic())
+                .flatMap(t -> listSubscriptions(azureProfile, t, credential.createTokenCredential(t), throwableList)).sequential().collectList()
+                .map(subscriptionsSet -> {
+                    for (List<Subscription> subscriptionsOnTenant : subscriptionsSet) {
+                        for (Subscription subscription : subscriptionsOnTenant) {
+                            String key = StringUtils.lowerCase(subscription.getId());
+                            subscriptionMap.putIfAbsent(key, subscription);
+                        }
+                    }
+                    for (Map.Entry<String, Throwable> tenantError : throwableList.entrySet()) {
```

Comment:
Shall we handle exception in `listSubscriptions(AzureProfile profile, String tenantId, TokenCredential credential, Map<String, Throwable> errorsOnTenant) `?